### PR TITLE
updates to noresm_develop to be close as possible to noresm2_3_develop

### PR DESCRIPTION
**Pull request by mvertens**
This PR updates the oslo_aero code base so that there are minimal differences between the branches noresm_develop and noresm2_3_develop. 

 The only files that are different between the two in src/ are:
```
oslo_aero_control.F90 
oslo_aero_nucleate_ice.F90 
oslo_aero_optical_params.F90 
```

This PR also resolved Issue #12 which now passes.

### CODE REVIEW

**Comment by oyvindseland** [comment]
Does this mean that the SE dycore also works in a 2.3 setting with Oslo-Aero or is it the development version pointing towards 2.5?

**Comment by mvertens** [comment]
@oyvindseland - there are two development branches of oslo-aero. One that goes into 2.3 and one that goes into 2.5. The actual oslo-aero code (in src/) is almost the same between the two except for one file with major differences. The shadow files (in src_cam/) are somewhat different - which is to be expected. You should be able to run the spectral element in 2.3 - but its not officially supported in that version. So @gold2718 will be extending the noresm2_3_develop branch of oslo-aero for the noresm2.3 release and I will be working on the noresm2_5_develop version for noresm2.5. Does this help clarify things?

**Comment by oyvindseland** [comment]
OK. Good to know

File: src/oslo_aero_coag.F90 [hunk 1]
```diff
@@ -639,13 +638,14 @@ subroutine clcoag(q, pmid, pdel, temperature, cldnum, cldfrc, delt_inverse, ncol
 
                    !process modes don't change mode except so4 condensate which becomes coagulate instead
                    !assumed to have same sink as MODE_IDX_OMBC_INTMIX_AIT
-                   if( .NOT. is_process_mode(l_index_donor,.true.)   &
-                        .OR. ( (l_index_donor.eq.chemistryIndex(l_so4_a1))  .AND. modeIndexCoagulator .eq. MODE_IDX_OMBC_INTMIX_COAT_AIT) ) then
+                   if( .NOT. is_process_mode(l_index_donor,.true.) .or.  &
+                        ( (l_index_donor == chemistryIndex(l_so4_a1)) .and. &
+                           modeIndexCoagulator == MODE_IDX_OMBC_INTMIX_COAT_AIT) ) then
```
Comment (gold2718) [inline src/oslo_aero_coag.F90:643]: There are parentheses around the first equality check but not the second one. Something to fix someday?

File: src/oslo_aero_coag.F90 [hunk 2]
```diff
@@ -659,15 +659,15 @@ subroutine clcoag(q, pmid, pdel, temperature, cldnum, cldfrc, delt_inverse, ncol
...
        do ispecie = 1, getNumberOfTracersInMode(coagulatingMode(iCoagulator))
           l_index_donor = getTracerIndex(coagulatingMode(iCoagulator) , ispecie ,.true.)
 
           !so4_a1 is a process mode (condensate), but is still lost in coagulation
-          if( .NOT. is_process_mode(l_index_donor, .true.)          &
-              .OR. ( (l_index_donor.eq.chemistryIndex(l_so4_a1)) .AND. coagulatingMode(iCoagulator) &
-              .eq. MODE_IDX_OMBC_INTMIX_COAT_AIT) ) then
+          if( .NOT. is_process_mode(l_index_donor, .true.) .or. &
+               ( (l_index_donor == chemistryIndex(l_so4_a1)) .and. &
+                  coagulatingMode(iCoagulator) == MODE_IDX_OMBC_INTMIX_COAT_AIT) ) then
```
Comment (gold2718) [inline src/oslo_aero_coag.F90:670]: There are parentheses around the first equality check but not the second one. Something to fix someday?

**Review by gold2718 — APPROVED** [review]
Looks good. Just a couple of comments about future cleanup

**Comment by mvertens** [comment]
@gold2718 - so I'm holding off a bit on this PR since the threading seems to be broken (ERP test crashes). I'm trying to track down if its the oslo_aero code or not. The PEM test passes. I'll raise issues once I sort this out.

File: src_cam/radiation.F90
Comment (gold2718) [inline src_cam/radiation.F90:1]: I prefer the old indentation even though that is not (yet?) used in the NorESM2.3 code.

**Review by gold2718 — COMMENTED** [review]
Would appreciate it if you would re-introduce indentation to radiation.F90 but otherwise looks good.

File: src_cam/radiation.F90
Comment (mvertens) [reply src_cam/radiation.F90:1]: Okay - I will reintroduce it.

**Pull request merged**